### PR TITLE
Laboratory work #2, Maria Nikitenko - 23FPL2

### PR DESCRIPTION
**Pull request by MariaNikitenko19**

### CODE REVIEW

File: lab_2_retrieval_w_bm25/main.py [hunk 1]
```diff
@@ -18,6 +18,11 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    text_only_words = re.sub('["|.|?|!|,|1|2|3|4|5|6|7|8|9|0|:|(|)]','',text.lower())
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:23]: woow. Can you explain it?

File: lab_2_retrieval_w_bm25/start.py [hunk 2]
```diff
@@ -30,5 +30,7 @@ def main() -> None:
     assert result, "Result is None"
 
 
+
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/start.py:33]: rm

File: lab_2_retrieval_w_bm25/start.py [hunk 3]
```diff
@@ -30,5 +30,7 @@ def main() -> None:
     assert result, "Result is None"
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/start.py:30]: Where is the using of your functions?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py (hunk 1 above)
Comment (MariaNikitenko19) [reply lab_2_retrieval_w_bm25/main.py]: I've done it in another way

File: lab_2_retrieval_w_bm25/start.py (hunk 2 above)
Comment (MariaNikitenko19) [reply lab_2_retrieval_w_bm25/start.py:52]: done

File: lab_2_retrieval_w_bm25/start.py (hunk 3 above)
Comment (MariaNikitenko19) [reply lab_2_retrieval_w_bm25/start.py:32]: done

File: lab_2_retrieval_w_bm25/main.py [hunk 4]
```diff
@@ -18,6 +19,13 @@ def tokenize(text: str) -> list[str] | None:
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(text, str):
+        return None
+    for symbol in text:
+        if not symbol.isalpha() and not symbol in ["-", " ","'"]:
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:25]: better use tuple and create it before loop

Comment (artyomtugaryov) [reply lab_2_retrieval_w_bm25/main.py:25]: `not symbol in ["-", " ","'"]` is mistake probably . Use `symbol not in ["-", " ","'"]`

File: lab_2_retrieval_w_bm25/settings.json [hunk 5]
```diff
@@ -1,3 +1,3 @@
 {
-    "target_score": 0
+    "target_score": 6
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/settings.json:2]: what about 8?

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py [hunk 6]
```diff
@@ -62,7 +89,19 @@ def calculate_tf(vocab: list[str], document_tokens: list[str]) -> dict[str, floa
...
+            or not all(isinstance(unique_word, str) for unique_word in vocab):
+        return None
+    if not isinstance(document_tokens, list)\
+            or not all(isinstance(token, str) for token in document_tokens):
+        return None
+    tf_dict = {}
+    for word in vocab:
+        if word in document_tokens:
+            tf_dict[word] = document_tokens.count(word) / len(document_tokens)
+        else:
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:102]: try to eliminate  `else`

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py [hunk 7]
```diff
@@ -33,6 +41,16 @@ def remove_stopwords(tokens: list[str], stopwords: list[str]) -> list[str] | Non
 
     In case of corrupt input arguments, None is returned.
     """
+    if not isinstance(tokens, list) or not isinstance(stopwords, list):
+        return None
+    if not all(isinstance(stopword, str) for stopword in stopwords)\
+            or not all(isinstance(token, str) for token in tokens):
+        return None
+    removed_stopwords = []
```
Comment (artyomtugaryov) [inline lab_2_retrieval_w_bm25/main.py:49]: can you do it in one line

**Review by artyomtugaryov — CHANGES_REQUESTED** [review]

File: lab_2_retrieval_w_bm25/main.py (hunk 4 above)
Comment (MariaNikitenko19) [reply lab_2_retrieval_w_bm25/main.py]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 6 above)
Comment (MariaNikitenko19) [reply lab_2_retrieval_w_bm25/main.py]: done

File: lab_2_retrieval_w_bm25/main.py (hunk 7 above)
Comment (MariaNikitenko19) [reply lab_2_retrieval_w_bm25/main.py]: done

**Pull request closed without merging**